prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -36,28 +36,30 @@ const val conditionExtensionurl = "https://reportstream.cdc.gov/fhir/StructureDe
 const val conditionCodeExtensionURL = "https://reportstream.cdc.gov/fhir/StructureDefinition/condition-code"
 
 /**
- * Looks up a condition code for the passed in [code] (typically a test) using the given [metadata] object
+ * Looks up condition codes for the provided LOINC [code] (typically a test) using the given [metadata] object
  * @param code test (or other type) code to look up a condition for
  * @param metadata metadata containing an observation-mapping lookup table
- * @return the condition code or null if no code was found
+ * @return a list of the matching condition codes (as `Coding`s)
  */
-private fun lookupCondition(code: Coding, metadata: Metadata): Coding? {
+private fun lookupConditionCodings(code: Coding, metadata: Metadata): List<Coding> {
     val mappingTable = metadata.findLookupTable("observation-mapping").also {
         if (it == null) { // could not load the table
             throw IllegalStateException("Unable to load lookup table 'observation-mapping' for condition stamping")
         }
     }!!
-    val condition = mappingTable.caseSensitiveDataRowsMap.find { // search for the code
+    val conditions = mappingTable.caseSensitiveDataRowsMap.filter { // find codes
         it[ObservationMappingConstants.TEST_CODE_KEY] == code.code
-    }
-    return if (condition.isNullOrEmpty()) { // could not find the code
-        null
-    } else { // code found; create Coding instance to return
-        Coding(
-            condition[ObservationMappingConstants.CONDITION_CODE_SYSTEM_KEY],
-            condition[ObservationMappingConstants.CONDITION_CODE_KEY],
-            condition[ObservationMappingConstants.CONDITION_NAME_KEY]
-        )
+    }.toList()
+    return if (conditions.isEmpty()) { // no codes found
```

Comment:
Nit: I think you can skip this conditional and just call `map` after `filter`?

Reply:
Adding to this for your functional programming knowledge:

Your standard list higher-order functions (`map`, `filter`, `fold`) will be no-ops on empty lists. In this instance if your `filter` returns an empty list your map function will do nothing. This is very handy for chaining together a bunch of functions without having to deal with conditionals.

Reply:
Yeah this was just a derp. Good catch thanks all 👌